src/InfinitePayCore.php

```diff
@@ -0,0 +1,385 @@
+<?php
+namespace Woocommerce\InfinitePay;
+
+if (!function_exists('add_action')) {
+    exit(0);
+}
+
+use Woocommerce\InfinitePay\Controller\Settings;
+use Woocommerce\InfinitePay\Controller\Checkout;
+use Woocommerce\InfinitePay\Helper\ApiInfinitePay;
+use Woocommerce\InfinitePay\Helper\Log;
+use Woocommerce\InfinitePay\Helper\Utils;
+use Woocommerce\InfinitePay\Helper\Constants;
+
+class InfinitePayCore extends \WC_Payment_Gateway
+{
+	public $core_settings;
+	public $api;
+
+    public function __construct()
+    {
+        $this->update_plugin_version();
+        $this->load_plugin_textdomain();
+        $this->setup_properties();
+		$this->init_form_fields();
+    
+		$this->core_settings = new Settings($this);
+        $this->title         = $this->core_settings->title;
+        $this->icon			 = $this->get_ip_icon();
+		$this->log			 = new Log($this);
+
+		$this->api = new ApiInfinitePay();
+
+		add_action('woocommerce_update_options_payment_gateways_' . $this->id, array($this, 'process_admin_options'));
+        add_action('wp_enqueue_scripts', array($this, 'payment_scripts'));
+        add_action('woocommerce_thankyou_' . $this->id, array($this, 'thank_you_page'));
+        add_filter('woocommerce_payment_complete_order_status', array($this, 'change_payment_complete_order_status'), 10, 3);
+        add_action('woocommerce_email_before_order_table', array($this, 'email_instructions'), 10, 3);
+    }
+
+    public static function load_plugin_textdomain()
+    {
+        $text_domain = 'infinitepay-woocommerce';
+        $locale      = apply_filters('plugin_locale', get_locale(), $text_domain);
+
+        $original_language_file = dirname(__FILE__) . '/../i18n/languages/infinitepay-woocommerce-' . $locale . '.mo';
+
+        unload_textdomain($text_domain);
+        load_textdomain($text_domain, $original_language_file);
+    }
+
+    public function update_plugin_version()
+    {
+        $old_version = get_option('_ip_version', '0');
+        if (version_compare(Constants::VERSION, $old_version, '>')) {
+            update_option('_ip_version', Constants::VERSION, true);
+        }
+    }
+
+    public static function woocommerce_instance()
+    {
+        if (function_exists('WC')) {
+            return WC();
+        } else {
+            global $woocommerce;
+
+            return $woocommerce;
+        }
+    }
+
+	public function admin_options()
+    {
+		?>
+		<h2>InfinitePay</h2>
+		<?php if (!$this->api->has_access_token): ?>
+			<div id="message" class="notice-warning notice">
+			<?php
+			printf(
+					'<p><strong>%s: </strong>%s <strong>%s</strong>.</p>',
+					__('Before accepting payments', 'infinitepay-woocommerce'),
+					__('you need to add your', 'infinitepay-woocommerce'),
+					__('Client ID and Client Secret', 'infinitepay-woocommerce')
```

Comment:
Please add `Access the <a href="https://money.infinitepay.io/settings/credentials">InfinitePay dashboard</a> to obtain your credentials` on this message

Reply:
Okey!